### ISSUE TITLE
Samsung UN46EH5300 not presenting any file | UMS 14.4.0 

### ISSUE THREAD

**Issue by pablosebastianu** [UniversalMediaServer/UniversalMediaServer#4956]
Hi,
I have a Samsung UN46EH5300. I've being updating UMS regularly but, at some point in time, I started to get the following error.
Please, find attach debug zip file.

From TV, I'm able to see the media folders, but when a select some of them, they are empty (not showing any file)

I'll appreciate your help

Thanks!

12:09:54 DEBUG Recognized media renderer "[TV]UN46EH5300"
12:09:54 DEBUG Bad id null found in path
12:09:54 DEBUG **Trying to browse direct children on a null object for objectID 'null' !**
12:10:16 DEBUG Finishing configuration for UnknownRenderer
12:10:16 DEBUG Updating status button for UnknownRenderer
12:10:16 DEBUG Using default media renderer "UnknownRenderer"
12:10:16 INFO  Media renderer was not recognized. Possible identifying HTTP headers:
User-Agent: Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/29.0.1547.66 Safari/537.36
X-device-id: 40:0D:10:2F:67:B3
[ums_dbg_2024-09-08-12-11.zip](https://github.com/user-attachments/files/16922715/ums_dbg_2024-09-08-12-11.zip)

**Comment by Mik-S-UMS** [comment]
For some reason UMS is sending the IDs as "null" to the TV. When the TV requests a file the ID is also null which creates an error.

```
CONTENT:
<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
    <s:Body>
        <u:BrowseResponse xmlns:u="urn:schemas-upnp-org:service:ContentDirectory:1">
            <Result><DIDL-Lite xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/" xmlns:dc="http://purl.org/dc/elements/1.1/" xm
<container id="null" childCount="3" parentID="0" restricted="1"><dc:title>Media Library</dc:title><res xmlns:dlna="urn:schemas-dln
<container id="null" childCount="1" parentID="0" restricted="1"><dc:title>Peliculas</dc:title><dc:date>2021-02-12T23:55:13</dc:dat
<container id="null" childCount="1" parentID="0" restricted="1"><dc:title>Crazy.Stupid.Love.2011.1080p.BluRay.x265-RARBG</dc:title
<container id="null" childCount="5" parentID="0" restricted="1"><dc:title>Web</dc:title><res xmlns:dlna="urn:schemas-dlna-org:meta
            <NumberReturned>4</NumberReturned>
            <TotalMatches>4</TotalMatches>
            <UpdateID>1</UpdateID>
        </u:BrowseResponse>
    </s:Body>
</s:Envelope>
```

```
Received a request from 192.168.0.6:
============================= UPNP HTTP SERVER REQUEST BEGIN ================================
POST /dev/a73261a1-6ccc-4c79-8d10-4e221056e4ed/svc/upnp-org/ContentDirectory/action HTTP/1.1

HEADER:
  Accept: */*
  User-Agent: SEC_HHP_[TV]UN46EH5300/1.0 DLNADOC/1.50
  Host: 192.168.0.3:5001
  SOAPACTION: "urn:schemas-upnp-org:service:ContentDirectory:1#Browse"
  Content-Type: text/xml; charset="utf-8"
  Content-Length: 454
  Connection: close

CONTENT:
<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
    <s:Body>
        <u:Browse xmlns:u="urn:schemas-upnp-org:service:ContentDirectory:1">
            <ObjectID>null</ObjectID>
            <BrowseFlag>BrowseDirectChildren</BrowseFlag>
            <Filter>*</Filter>
            <StartingIndex>0</StartingIndex>
            <RequestedCount>1</RequestedCount>
            <SortCriteria/>
        </u:Browse>
    </s:Body>
</s:Envelope>

============================= UPNP HTTP SERVER REQUEST END ==================================
Bad id null found in path
Trying to browse direct children on a null object for objectID 'null' !
```

I did see an error on the database upgrade that may be the cause.
```
INFO  2024-09-08 12:07:27.197 [main] Upgrading database "medias" table "VIDEO_METADATA" from version 4 to 7
ERROR 2024-09-08 12:07:27.248 [main] Error checking tables: Column "VIDEO_METADATA_POSTERS.FILEID" not found; SQL statement:
UPDATE VIDEO_METADATA SET VIDEO_METADATA.POSTER = (SELECT VIDEO_METADATA_POSTERS.POSTER FROM VIDEO_METADATA_POSTERS WHERE VIDEO_METADATA_POSTERS.FILEID = VIDEO_METADATA.FILEID LIMIT 1) [42122-230]
TRACE 2024-09-08 12:07:27.249 [main] 
org.h2.jdbc.JdbcSQLSyntaxErrorException: Column "VIDEO_METADATA_POSTERS.FILEID" not found; SQL statement:
UPDATE VIDEO_METADATA SET VIDEO_METADATA.POSTER = (SELECT VIDEO_METADATA_POSTERS.POSTER FROM VIDEO_METADATA_POSTERS WHERE VIDEO_METADATA_POSTERS.FILEID = VIDEO_METADATA.FILEID LIMIT 1) [42122-230]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:514)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:489)
	at org.h2.message.DbException.get(DbException.java:223)
	at org.h2.message.DbException.get(DbException.java:199)
	at org.h2.expression.ExpressionColumn.getColumnException(ExpressionColumn.java:244)
	at org.h2.expression.ExpressionColumn.optimizeOther(ExpressionColumn.java:226)
	at org.h2.expression.ExpressionColumn.optimize(ExpressionColumn.java:213)
	at org.h2.expression.condition.Comparison.optimize(Comparison.java:147)
	at org.h2.expression.Expression.optimizeCondition(Expression.java:148)
	at org.h2.command.query.Select.prepareExpressions(Select.java:1235)
	at org.h2.command.query.Query.prepare(Query.java:232)
	at org.h2.expression.Subquery.optimize(Subquery.java:98)
	at org.h2.command.dml.SetClauseList$SetSimple.mapAndOptimize(SetClauseList.java:409)
	at org.h2.command.dml.SetClauseList.mapAndOptimize(SetClauseList.java:219)
	at org.h2.command.dml.Update.doPrepare(Update.java:130)
	at org.h2.command.dml.DataChangeStatement.prepare(DataChangeStatement.java:37)
	at org.h2.command.Parser.prepareCommand(Parser.java:489)
	at org.h2.engine.SessionLocal.prepareLocal(SessionLocal.java:644)
	at org.h2.engine.SessionLocal.prepareCommand(SessionLocal.java:560)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1164)
	at org.h2.jdbc.JdbcStatement.executeUpdateInternal(JdbcStatement.java:190)
	at org.h2.jdbc.JdbcStatement.executeUpdate(JdbcStatement.java:147)
	at com.zaxxer.hikari.pool.ProxyStatement.executeUpdate(ProxyStatement.java:119)
	at com.zaxxer.hikari.pool.HikariProxyStatement.executeUpdate(HikariProxyStatement.java)
	at net.pms.database.DatabaseHelper.executeUpdate(DatabaseHelper.java:630)
	at net.pms.database.MediaTableVideoMetadata.upgradeTable(MediaTableVideoMetadata.java:237)
	at net.pms.database.MediaTableVideoMetadata.checkTable(MediaTableVideoMetadata.java:170)
	at net.pms.database.MediaDatabase.checkTables(MediaDatabase.java:89)
	at net.pms.database.MediaDatabase.onOpening(MediaDatabase.java:56)
	at net.pms.database.Database.init(Database.java:98)
	at net.pms.database.MediaDatabase.init(MediaDatabase.java:187)
	at net.pms.PMS.init(PMS.java:260)
	at net.pms.PMS.createInstance(PMS.java:1007)
	at net.pms.PMS.main(PMS.java:958)
```

**Comment by pablosebastianu** [comment]
Hi Mik-S-UMS,
Thanks for your feedback.
I just re-installed UMS, but the issue is still there.

Please, find attached, debug zip file I just get.

BR
[ums_dbg_2024-09-08-14-48.zip](https://github.com/user-attachments/files/16923133/ums_dbg_2024-09-08-14-48.zip)

**Comment by Mik-S-UMS** [comment]
This has the same error so there is a problem with the database update. 

Deleting the database "C:\ProgramData\UMS\database\medias.mv.db" (or renaming it) will allow UMS to rebuild it from scratch and this should fix it.

I saw that some extra checks were added recently to the database upgrade to prevent things like this happening (I think it was #4947) but they would not have made it into any releases yet.
